zebra/zebra_rib.c

```diff
@@ -2745,6 +2748,24 @@ int rib_add_multipath(afi_t afi, safi_t safi, struct prefix *p,
 		if (CHECK_FLAG(same->status, ROUTE_ENTRY_REMOVED))
 			continue;
 
+		/* If kernel_reconcile is set, then mark Zebra-originated route
+		 * for deletion. Zebra learned these routes from kernel while
+		 * start up. Now, since we learned new route for same destination,
+		 * this is the time to clear stale route, so that both kernel
+		 * and FPM are updated while rib_process. This will result in no
+		 * traffic loss, if same route->NH is learned after startup.
+		 */
+		if (kernel_reconcile
```

Comment:
Once kernel_reconcile_rt has become Non NULL via same, Can we bypass this check ?

Reply:
The if conditions includes "kernel_reconcile" which is set by -K option, I want to run my code only when user starts Zebra with -K option. "kernel_reconcile_rt" is local variable and will have no impact on this condition. Thanks.

Reply:
I understand that you want to execute the code only when kernel_reconcile flag is set to True. My comment is not about that. While iterating in this loop, and when kernel_reconcile_rt is set to same and non-NULL, you don't need to perform this check. Is this right ? if not you will overwrite kernel_reconcile_rt every time with same object. 

So my comment is that, Can you add a non-null check for kernel_reconcile_rt also into the above check so that control doesn't go inside this "if loop" every time.

Reply:
Got it, There is no harm in adding this condition. 

As I know Linux kernel returns to zebra one route per prefix., because kernel does not install more than one route per prefix from zebra. So once we hit this condition, it should not hit again. But yeah in that case, we avoid condition all together by checking kernel_reconcile_rt. Thanks.